### ISSUE TITLE
lint variable assignments that could be part of a method/field chain

### ISSUE THREAD

**Issue by oli-obk** [rust-lang-nursery/rust-clippy#2035]
```rust
let x = foo.bar;
let y = x.met();
```

Should be

```rust
let y = foo.bar.met();
```

If x is not used anywhere else

**Comment by llogiq** [comment]
I'd argue that this lint should be `allow`, if only for the fact that introducing a reasonably a named binding can help understand the code.

Also one needs to be careful if the type is given.

**Comment by oli-obk** [comment]
> Also one needs to be careful if the type is given.

Yea, if there's a type, this should not lint

> I'd argue that this lint should be allow, if only for the fact that introducing a reasonably a named binding can help understand the code.

How about not linting if the name is longer than 3 chars?
We could also lint if the variable name is the same as a function or field in the method chain leading to the value of the variable, since that would then be redundant (like if the variable `x` were named `bar` in the above example).

**Comment by llogiq** [comment]
I'm ok with linting if the name doesn't add anything but repetition to the code. The heuristics you propose seem sensible. I'd still like to see it on a few crates before making it `warn` by default. I have a hunch that this could be *very* trigger happy on normal code, drowning out other lints.

**Comment by oli-obk** [comment]
> The heuristics you propose seem sensible. I'd still like to see it on a few crates before making it warn by default. I have a hunch that this could be very trigger happy on normal code, drowning out other lints.

Perfectly fine with me, as long as we have a path towards making it warn by default

**Comment by llogiq** [comment]
By the way, how's that clippy-bomb project commencing? :smile:

**Comment by oli-obk** [comment]
Haha, yea, forgot about that. We can try getting it into cargobomb
